lib/ui/common/ActionSheet.dart

```diff
@@ -0,0 +1,146 @@
+  final Color backgroundColor;
+  final Color indicatorLineColor;
+  final Color confirmButtonBackgroundColor;
+  final Color cancelButtonBackgroundColor;
+
+  final EdgeInsets indicatorLineEdgePadding;
+
+  final Radius cornerRadius;
+  final Radius indicatorLineRadius;
+
+  final double indicatorLineThickness;
+  final double indicatorLineWidth;
+
+  ActionSheetStyle(
+      {this.backgroundColor,
+      this.indicatorLineColor,
+      this.confirmButtonBackgroundColor,
+      this.cancelButtonBackgroundColor,
+      this.indicatorLineEdgePadding,
+      this.cornerRadius,
+      this.indicatorLineRadius,
+      this.indicatorLineThickness,
+      this.indicatorLineWidth});
+
+  factory ActionSheetStyle.defaultStyle() {
+    return ActionSheetStyle(
+      backgroundColor: const Color(0xFFffffff),
+      indicatorLineColor: const Color(0xFFe0e1ee),
+      confirmButtonBackgroundColor: const Color(0xFF24d900),
+      cancelButtonBackgroundColor: const Color(0xFFe9eaf2),
+      indicatorLineEdgePadding: const EdgeInsets.only(top: 8, bottom: 50),
+      cornerRadius: const Radius.circular(40),
+      indicatorLineRadius: const Radius.circular(2.5),
+      indicatorLineThickness: 5,
+      indicatorLineWidth: 40,
+    );
+  }
+
+  ActionSheetStyle copyWith(
+      {Color backgroundColor,
+      Color indicatorLineColor,
+      Color confirmButtonBackgroundColor,
+      Color cancelButtonBackgroundColor,
+      EdgeInsets indicatorLineEdgePadding,
+      Radius cornerRadius,
+      Radius indicatorLineRadius,
+      double indicatorLineThickness,
+      double indicatorLineWidth}) {
+    return ActionSheetStyle(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      indicatorLineColor: indicatorLineColor ?? this.indicatorLineColor,
+      confirmButtonBackgroundColor:
+          confirmButtonBackgroundColor ?? this.confirmButtonBackgroundColor,
+      cancelButtonBackgroundColor:
+          cancelButtonBackgroundColor ?? this.cancelButtonBackgroundColor,
+      indicatorLineEdgePadding:
+          indicatorLineEdgePadding ?? this.indicatorLineEdgePadding,
+      cornerRadius: cornerRadius ?? this.cornerRadius,
+      indicatorLineRadius: indicatorLineRadius ?? this.indicatorLineRadius,
+      indicatorLineThickness:
+          indicatorLineThickness ?? this.indicatorLineThickness,
+      indicatorLineWidth: indicatorLineWidth ?? this.indicatorLineWidth,
+    );
+  }
+}
+
+class ActionSheet extends StatelessWidget {
+  final ActionSheetViewModel _viewModel;
+  final ActionSheetStyle _style;
+  final ActionSheetListItemStyle _cellStyle;
+
+  const ActionSheet(
+      {Key key,
+      ActionSheetViewModel viewModel,
+      ActionSheetStyle style,
+      ActionSheetListItemStyle cellStyle})
+      : this._viewModel = viewModel,
+        this._style = style,
+        this._cellStyle = cellStyle,
+        super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        color: _style.cornersColor,
```

Comment:
have this as a constant in the widget. The Style class is about all the things you want external users to be able to change. since this is constant, lets not expose to them.

Reply:
Done